docs/hugo/content/design/ADR-2022-09-Multiple-Credential-Operator.md

```diff
@@ -0,0 +1,153 @@
+---
+title: '2022-09: Support For Multiple Credentials Under Global Operator'
+---
+
+## Context
+
+The operator needs to support creating Azure resources using multiple credentials, managed by different groups, within one cluster by a single global instance of operator. 
+The idea is to support the above by having **Namespaced secret**, **Per-resource-group secret** and Per-resource secret. 
+
+### Namespaced Secret
+
+Namespaced secret holds credentials used for the resources in a namespace.
+
+### Per-resource-group Secret
+
+Per-resource-group secret holds credential used for the resource under a resource group.
+
+### Per-resource Secret
+
+Per-resource-group secret holds credential used for the resource separately.
+
+## Credential selection hierarchy
+
+We'll be using a pattern to determine which of the above secrets we use for operation on a resource. As in the below flow chart, if a resource is applied to the operator and the resource exists, operator would fetch the credentials from
+cache and perform the operation. If its a new resource, operator would have to go through the secret selection hierarchy. Where, operator would first check if Per-resource or per-resource-group secret exists, if not, then will check for 
+the namespaced secret. If any of the above is provided, operator would perform actions according to the **options** below. If none is provided, we'll fallback to use the global credential for that resource. 
+
+![hierarchy](images/adr-2022-09-multiple-credential-operator.png) 
+
+## Options
+
+In all the proposed solutions, we would need to cache the credentials to be used for resources when we reconcile and a mechanism to identify the credentials used for a resource. 
+
+### Option 1: Secrets lookup
+
+In secrets lookup option, we would need a fixed secret pattern like below. Which then will be looked up for each time a new resource is created. Operator would look up for the secret pattern for resource group first, then namespace. If secret exists, cache the credentials for the resource to be used for the next time or use the default global.
```

Comment:
Grammar tip - generally want to avoid using "which" as a linking word like this.

```suggestion
In secrets lookup option, fixed naming patterns would be used to look up a secret each time a new resource is created. Operator would first use the naming pattern for the resource group, then for the cluster namespace. If a secret is found with the expected name, the credentials for the resource would be cached for use the next time. If no secret was found matching any patter, the default global secret would be used.
```